### PR TITLE
Fix charts with numeric dimensions

### PR DESCRIPTION
**Pull request by krakov**

### CODE REVIEW

File: src/streamlit_app.py [hunk 1]
```diff
@@ -382,106 +383,114 @@ def render_grouped_bar_chart(
...
+        df_to_show[col] = df[col].astype(str)
+
+    # date dimension
     if len(date_columns) > 0:
         df_to_show = df_to_show.rename(columns={date_columns[0]: "index"}).set_index(
             "index",
         )
 
-        if len(str_columns) == 0:
+        if len(dim_columns) == 0:
```
Comment (baruchoxman) [inline src/streamlit_app.py:446]: Did you test this case on the new code?

File: src/streamlit_app.py [hunk 2]
```diff
@@ -382,106 +383,114 @@ def render_grouped_bar_chart(
...
+            return False
 
-    # text
-    elif len(str_columns) >= 1:
-        metric_column = numeric_columns[0]  # First numeric column
+    # dimensions
+    elif len(dim_columns) >= 1:
+        metric_column = metric_columns[0]  # First numeric column
 
         if len(df_to_show) > 50:
```
Comment (baruchoxman) [inline src/streamlit_app.py:457]: Shouldn't we apply this in all cases? Now it's not covered in the date columns flow

**Review by baruchoxman — APPROVED** [review]

File: src/streamlit_app.py (hunk 2 above)
Comment (krakov) [reply src/streamlit_app.py:457]: no as it handles very well compression of time series

Comment (krakov) [reply src/streamlit_app.py:457]: and sorting there would be wrong

File: src/streamlit_app.py (hunk 1 above)
Comment (krakov) [reply src/streamlit_app.py:446]: yes, "profit monthly"

**Pull request merged**